api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -137,6 +137,11 @@ message OAuth2Config {
   // If this value is not set, it will default to ``0s``. In this case, the expiry must be set by
   // the authorization server or the OAuth flow will fail.
   google.protobuf.Duration default_expires_in = 13;
+
+  // Requests that match any of the provided matchers will immediately return unauthorized response
+  // when tokens are not valid (instead of redirecting to authorization server as usual).
+  // This behavior is useful for AJAX and asset requests to avoid the OAuth flood.
+  repeated config.route.v3.HeaderMatcher ajax_request_matcher = 14;
```

Comment:
I don't know much about this stuff but are there other cases where it would be desirable to have this behavior? I.e., is this AJAX specific? Would it be better to just call this `immediate_unauthorized_response_matcher` or something like that?